### PR TITLE
Add two new build status : with warning and with error

### PR DESCRIPTION
**Pull request by flomillot**

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]
Code OK

File: src/components/graph/menus/create-node-menu.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ const CreateNodeMenu = ({
             action: () => buildNode(),
             id: 'buildNode',
             disabled:
-                activeNode?.data?.buildStatus === 'BUILT' ||
+                activeNode?.data?.buildStatus.startsWith('BUILT') ||
```
Comment (FranckLecuyer) [inline src/components/graph/menus/create-node-menu.js:115]: Maybe ?.startsWith('BUILT') for safety ?

File: src/components/network-modification-tree.js [hunk 2]
```diff
@@ -77,8 +77,10 @@ const NetworkModificationTree = ({
             ) {
                 case 'BUILT':
                     return '#70d136';
-                case 'NOT_BUILT':
-                    return '#9196a1';
+                case 'BUILT_WITH_WARNING':
```
Comment (FranckLecuyer) [inline src/components/network-modification-tree.js:80]: For the colors, you can use the constants defined in network-modification-node.js

File: src/components/graph/util/model-functions.js [hunk 3]
```diff
@@ -73,7 +74,7 @@ export function isNodeReadOnly(node) {
 export function isNodeBuilt(node) {
     if (!node) return false;
     if (node.type === 'ROOT') return true;
-    return node.data?.buildStatus === 'BUILT';
+    return node.data?.buildStatus.startsWith('BUILT');
```
Comment (FranckLecuyer) [inline src/components/graph/util/model-functions.js:77]: Maybe ?.startsWith('BUILT') for safety ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/graph/nodes/network-modification-node.js [hunk 4]
```diff
@@ -117,6 +126,19 @@ const NetworkModificationNode = (props) => {
             : 'unset';
     };
 
+    function getClassForBanner(buildStatus) {
+        switch (buildStatus) {
+            case 'BUILT':
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/graph/nodes/network-modification-node.js:131]: Definitely, I'm doing it

File: src/components/network-modification-tree.js (hunk 2 above)
Comment (flomillot) [reply src/components/network-modification-tree.js:80]: Actually the colors was not the same that's why I didn't use constants, but IMO we should use the same everywhere

File: src/components/graph/util/model-functions.js (hunk 3 above)
Comment (flomillot) [reply src/components/graph/util/model-functions.js:77]: Done

File: src/components/graph/menus/create-node-menu.js [hunk 5]
```diff
@@ -176,7 +176,7 @@ const CreateNodeMenu = ({
             id: 'exportCaseOnNode',
             disabled:
                 activeNode?.type !== 'ROOT' &&
-                activeNode?.data?.buildStatus !== 'BUILT',
+                !activeNode?.data?.buildStatus.startsWith('BUILT'),
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/graph/menus/create-node-menu.js:179]: Done

File: src/components/graph/menus/create-node-menu.js (hunk 1 above)
Comment (flomillot) [reply src/components/graph/menus/create-node-menu.js:115]: Done

File: src/components/network-modification-tree.js (hunk 2 above)
Comment (laurgarn) [reply src/components/network-modification-tree.js:80]: I'd like this aspect to sneak in a theme for third mode after daylight and night modes

**Review by FranckLecuyer — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Good job !

**Pull request merged**